StudentList/Domain/Reducers/StudentListReducer.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+using StudentList.Domain.Actions;
+using StudentList.Models;
+
+namespace StudentList.Domain.Reducers
+{
+    public static class StudentListReducer
+    {
+        public static IEnumerable<Student> Reduce(IEnumerable<Student> state, object action)
+        {
+            switch (action)
+            {
+                case StudentReceived studentReceivedAction:
+                    return Reduce(studentReceivedAction);
+                case StudentListChanged studentListUpdated:
+                    return null;
+                default:
+                    return state;
+            }
+        }
+
+        private static IEnumerable<Student> Reduce(StudentReceived studentReceivedAction)
+        {
+            return studentReceivedAction.StudentList;
```

Comment:
You should always project results before saving to state, so you ensure that in state you have NEW objects and not same objects as in action. Also, Student in state must be immutable, and you just reuse your Mutable student class, fix it.